src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package org.generationcp.middleware.api.germplasm;
+
+import org.apache.commons.lang.StringUtils;
+import org.generationcp.middleware.domain.germplasm.GermplasmNameRequestDto;
+import org.generationcp.middleware.hibernate.HibernateSessionProvider;
+import org.generationcp.middleware.manager.DaoFactory;
+import org.generationcp.middleware.pojos.Name;
+import org.generationcp.middleware.pojos.UDTableType;
+import org.generationcp.middleware.pojos.UserDefinedField;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.Collections;
+import java.util.List;
+
+@Service
+@Transactional
+public class GermplasmNameServiceImpl implements GermplasmNameService {
+
+	private final DaoFactory daoFactory;
+
+	public GermplasmNameServiceImpl(final HibernateSessionProvider sessionProvider) {
+		this.daoFactory = new DaoFactory(sessionProvider);
+	}
+
+	@Override
+	public Name getNameByNameId(final Integer nameId) {
+		return daoFactory.getNameDao().getById(nameId);
+	}
+
+	@Override
+	public Name getPreferredName(final Integer gid) {
+		final List<Name> names = daoFactory.getNameDao().getByGIDWithListTypeFilters(gid, 1, null);
+		if (!names.isEmpty()) {
+			return names.get(0);
+		}
+		return null;
+	}
+
+	@Override
+	public UserDefinedField getNameType(final String nameTypeCode) {
```

Comment:
In GermplasmService we already have
`	List<GermplasmNameTypeDTO> filterGermplasmNameTypes(Set<String> codes);`
You can use this one so we avoid code duplication

Reply:
done.